Update the Audio track in row, box and full layouts

obulat opened the pull request:
<!-- prettier-ignore -->
## Fixes
<!-- If PR doesn't fully resolve the issue, replace 'Fixes' below with 'Related to'. -->
<!-- If there is no issue being resolved, please consider opening one before creating this pull request. -->

Fixes #2910 by @fcoveram
Fixes #2318 by @sarayourfriend 

## Description
<!-- Concisely describe what the pull request does. -->
<!-- Add screenshots, videos, or other media to show the problem and the solution when appropriate. -->
This PR updates the audio track layouts to match the updated designs.

## Testing Instructions
<!-- Give steps for the reviewer to verify that this PR fixes the problem; or delete this section entirely. -->


## Checklist
<!-- Replace  the [ ] with [x] to check the boxes. -->

- [ ] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [ ] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin
<!-- You must read and understand the following attestation. -->

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

obulat commented:
@fcoveram, the row layout breaks on `sm` breakpoint when the creator and the title are long:
<img width="406" alt="Screenshot 2023-11-07 at 8 58 50 AM" src="https://github.com/WordPress/openverse/assets/15233243/690d24cf-3b4d-47c5-a0c2-9d275aeb0f04">
How should it be fixed?

fcoveram commented:
For the large variant, both fields should be trimmed by showing `…` at the end, as in the current version.

<img src="https://github.com/WordPress/openverse/assets/895819/dd496cf9-09b3-4513-b200-0209f5f0f6fb" alt="Screenshot of row large component">

For the medium variant in `sm`, I think we can try the following:
* Trim the title as in large variant
* Hide the license name, as the component's smallest version does
* Trim the creator's field based on the remaining space

I added the two examples in the [Components section](https://www.figma.com/file/LMGuwErTWIYkOIEZI4t320/Audio-components-improvement?type=design&node-id=1244-57599&mode=design&t=SINVCXVjcXKryCQl-11).

What do you think?

![Design of audio component in large and medium variants](https://github.com/WordPress/openverse/assets/895819/b85df908-0fde-4d81-9862-9047e7cce149)



---

On the other hand, I noticed a spacing inconsistency in the row variant that was already fixed and updated in both design file and library. The space between audio info and control is now `24px (6)`.

![CleanShot 2023-11-07 at 09 43 54@2x](https://github.com/WordPress/openverse/assets/895819/f3b0457a-5638-4791-97ce-4cdcfdd961d7)

obulat commented:
I added the "ellipsis"('...') to title and the creator line in the `large` and `medium` variants, removed the license name in `medium` variant for breakpoints under `md` (so, `sm`) and increased the spacing before the audio control, @fcoveram.

fcoveram reviewed (changes requested):
## Box

**Note:** In the current storybook documentation, we have "medium" and "large" variants. But in this update we don’t need the large one as both are the same.

- In the small variant the wave area still shows up. In the new design this part is replaced with the category field. [Screenshot](https://share.cleanshot.com/GnntCXCx).
- In the medium variant, there is a gap of 8px between audio control and wave area that should not exist. [Screenshot](https://share.cleanshot.com/9QWbV4P6). 
- In small and medium variants, audio control does not have the hover style applied.

I also noticed that I can not control the audio playing with `← / →` keys on my keyboard to move forward and backward in time. This is not a style issue but sharing it in case it happens to anyone else as well. If so, I can create a ticket reporting it.

## Row

- In medium variant, the gap between cover image and audio should be 24px. There is a margin left that adds unnecessary space. [Screenshot](https://share.cleanshot.com/9BcvMZHg)
- In large variant, the gap between cover image and audio should be 16px. [Screenshot](https://share.cleanshot.com/b8CYJgpY).
- In large variant, there is an extra space on audio control's left side that adds unnecessary space. [Screenshot](https://share.cleanshot.com/mPjNZcMJ).
- In large variant, I noticed that audio tracks without the category field keep the assigned space and it creates an empty spot. We can try removing both the space and the dot to align all text to the left. In the small variant looks great. [Screenshot](https://share.cleanshot.com/xlZpfMLH).

## Full

Everything looks correct 🙂

---

Excellent work Olga, this update carries many changes and all the observations described above are subtle tweaks 👏✨

obulat commented:
Thank you for the detailed and kind review, @fcoveram! 

Commenting on the requested changes.

> ## Box
> **Note:** In the current storybook documentation, we have "medium" and "large" variants. But in this update we don’t need the large one as both are the same.

I updated the box layout to only have the "large" and "small" variant (instead of "medium" and "small"). Would that be OK?

> * In the small variant the wave area still shows up. In the new design this part is replaced with the category field. [Screenshot](https://share.cleanshot.com/GnntCXCx).

Apparently, I wasn't passing the `size` to the box layout, which caused some weird things. Updated, so now the screens below `sm` show the `small` box layout with the category in place of the waveform, and the larger screens use the `large` sized box layout.

> * In the medium variant, there is a gap of 8px between audio control and wave area that should not exist. [Screenshot](https://share.cleanshot.com/9QWbV4P6).

Removed.

> * In small and medium variants, audio control does not have the hover style applied.

Added the `transparent-dark` style for the `VAudioControl` in box layout - now the button on hover changes color. **Is there any other change that should be applied to the box audio track on hover?**

> I also noticed that I can not control the audio playing with `← / →` keys on my keyboard to move forward and backward in time. This is not a style issue but sharing it in case it happens to anyone else as well. If so, I can create a ticket reporting it.

I remember discussing somewhere that the box layout's waveform should not be interactive. So, the users can _see_ the waveform (and its progress when the audio is playing), but cannot interact with it, neither with the arrow keys nor with the mouse. **Should we make the waveform interactive for box layout, too?**

> ## Row
> * In medium variant, the gap between cover image and audio should be 24px. There is a margin left that adds unnecessary space. [Screenshot](https://share.cleanshot.com/9BcvMZHg)
> * In large variant, the gap between cover image and audio should be 16px. [Screenshot](https://share.cleanshot.com/b8CYJgpY).
> * In large variant, there is an extra space on audio control's left side that adds unnecessary space. [Screenshot](https://share.cleanshot.com/mPjNZcMJ).

I used the CSS grid for row layout, and set the horizontal gap to the smallest size, which is 16px. Then, I added padding between the audio control and the audio data in the `large` variant of the row layout. Previously, I tried implementing it in this way, but did it incorrectly (also applying to the `medium` row, and adding the additional padding to both columns - fixed now)


> * In large variant, I noticed that audio tracks without the category field keep the assigned space and it creates an empty spot. We can try removing both the space and the dot to align all text to the left. In the small variant looks great. [Screenshot](https://share.cleanshot.com/xlZpfMLH).

Added a check to the `dot`: display it in the `large` row layout only if the audio has a category.

fcoveram commented:
I didn't refresh the page so I couldn't see the review request notification. Reviewing it now

fcoveram reviewed (approved):
It looks great 🚀
